### PR TITLE
feat: Fixing problems with generated code for golang and adding sample manifest for testing.

### PR DESCRIPTION
**Pull request by dominikhaska**
## This PR

Several small changes:
- Add sample manifest
- Fix integer parsing and generating for golang
- Small fixes for missing defaults in switches

### Related Issues
No issue - I've just tried to use the code to generate accessors

### CODE REVIEW

File: sample/golang/golang_sample.json
Comment (beeme1mr) [inline sample/golang/golang_sample.json:1]: Could this be moved up a folder since it doesn't seem to be Go-specific?

Comment (dominikhaska) [reply sample/golang/golang_sample.json:1]: yes - make sense to move it up

Comment (dominikhaska) [reply sample/golang/golang_sample.json:1]: done

**Review by beeme1mr — APPROVED** [review]

**Comment by beeme1mr** [comment]
Thanks! I'll wait for @anghelflorinm to have a look. Feel free to merge when you're ready.

**Review by anghelflorinm — APPROVED** [review]

**Pull request merged**